### PR TITLE
Try to fix #6, IndexError

### LINKED ISSUE #6: Index out of bounds exception.
So this happened:

```
00:27:28 - ERROR - shift_helper.Check | list index out of range
Traceback (most recent call last):
  File "/Users/Kai/anaconda3/envs/shift/lib/python3.5/site-packages/shifthelper/checks/__init__.py", line 51, in run
    self.check()
  File "/Users/Kai/anaconda3/envs/shift/lib/python3.5/site-packages/shifthelper/checks/webdim.py", line 11, in check
    s = smart_fact_crawler.status()
  File "/Users/Kai/anaconda3/envs/shift/lib/python3.5/site-packages/smart_fact_crawler/__init__.py", line 182, in status
    'Time_Stamp': smartfact_time2datetime(tc[0, 0]),
  File "/Users/Kai/anaconda3/envs/shift/lib/python3.5/site-packages/smart_fact_crawler/__init__.py", line 23, in __getitem__
    item = self.page_payload[row].split()[col]
IndexError: list index out of range
```

### PR DESCRIPTION
**Pull request by relleums**
If an IndexError is raised while accessing the TableCrawler, most likely because the smart fact web page is corrupted, the TableCrawler will try to reload the page just as it is done on connection problems. After 10 trails, finally an exception is raised.

### CODE REVIEW

File: smart_fact_crawler/__init__.py [hunk 1]
```diff
@@ -45,6 +46,18 @@ def _build_page_payload(self):
             for line in self.web_page.text.splitlines()
         ]
 
+    def _load_payload(self):
+        self._request_web_page()
+        self._build_page_payload()
+
+    def _acknowledge_error_and_wait_a_moment(self):
+        print('acknowledge_error')
```
Comment (MaxNoe) [inline smart_fact_crawler/__init__.py]: No prints please

File: smart_fact_crawler/__init__.py [hunk 2]
```diff
@@ -13,30 +13,31 @@
...
     def __getitem__(self, index):
         row, col = index
-        item = self.page_payload[row][col]
+        while True:
+            try:
+                item = self.page_payload[row][col]
+            except IndexError:
+                self._acknowledge_error_and_wait_a_moment()
+            else:
+                self.connection_error_counter = 0
```
Comment (dneise) [inline smart_fact_crawler/__init__.py]: could the else part also be in the try part?

File: smart_fact_crawler/__init__.py [hunk 3]
```diff
@@ -13,30 +13,31 @@
...
-                self.connection_error_counter[url] += 1
-                if self.connection_error_counter[url] >= 10:
-                    raise
-                else:
-                    # sleep between 1 and 2 seconds.
-                    time.sleep(1. + random.random())
+                self._acknowledge_error_and_wait_a_moment()
             else:
-                self.connection_error_counter[url] = 0
+                self.connection_error_counter = 0
```
Comment (dneise) [inline smart_fact_crawler/__init__.py]: could the else part also be in the try part?

File: smart_fact_crawler/__init__.py [hunk 4]
```diff
@@ -45,6 +46,17 @@ def _build_page_payload(self):
             for line in self.web_page.text.splitlines()
         ]
 
+    def _load_payload(self):
```
Comment (dneise) [inline smart_fact_crawler/__init__.py]: maybe move this up below the __init__?

**Pull request merged**